### PR TITLE
Randomevents

### PR DESCRIPTION
**Pull request by GraysonW99**
Includes the change to include the current system in the list of reachable systems

Bug fixes to market logic:
Fixed a bug in the initialize inventory method in the marketplace viewmodel where the wrong variable was used to check whether the TTP matched and the amount of a good in the market should be boosted.
Fixed a bug where isTTP in tradeGood class was checking if it was <= and not ==

Added logic for random market events:
Added a method to planet class to roll for a random event and set it to the market's event.
Called the new method in location viewmodel after updating location in travelToPlanet method

### CODE REVIEW

File: app/src/main/java/com/communistutopia/spacetrader/viewmodel/MarketplaceViewModel.kt [hunk 1]
```diff
@@ -69,14 +69,19 @@ class MarketplaceViewModel : ViewModel() {
...
     /**
      * Goods that meet the tech level which produces the most are initialized to 5 * the base value.
-     *
+     * Goods that do not meet the MTLP are kept at 0
+     * The rest are set to the base amount
      */
     fun initializeInventory() {
         for (entry in player.value!!.location.market.inventory) {
-            if (entry.value.isMTLP(player.value!!.location.market)) {
+            if (entry.value.isTTP(player.value!!.location.market)) {
```
Comment (rohanrk) [inline app/src/main/java/com/communistutopia/spacetrader/viewmodel/MarketplaceViewModel.kt:77]: TTP only dictates if the player can sell to the planet right? Or am I mistaken?

File: app/src/main/java/com/communistutopia/spacetrader/viewmodel/LocationViewModel.kt [hunk 2]
```diff
@@ -21,6 +21,7 @@ class LocationViewModel: ViewModel() {
         if(player.value!!.spaceship.canTravelTo(player.value!!.system, destinationSystem)) {
             var dist = player.value!!.system.getDistance(destinationSystem)
             updateLocation(destinationSystem, destinationPlanet)
+            player.value!!.location.rollForRandomEvent()
```
Comment (rohanrk) [inline app/src/main/java/com/communistutopia/spacetrader/viewmodel/LocationViewModel.kt:33]: Just a note, we should probably have a toast or label somewhere in the MarketView displaying if this event triggers. @DrakeWitt and I could probably look into that.

**Comment by bayne-grubb** [comment]
Yee I'm working on that part. Was going to finish it this morning but am
dead so will probably finish it later tn

File: app/src/main/java/com/communistutopia/spacetrader/model/Player.kt [hunk 3]
```diff
@@ -16,21 +15,41 @@ import kotlinx.android.parcel.Parcelize
...
     var credits: Int = 1000,
     var charName: String = "",
     var pilotSkill: Int = 0,
     var fighterSkill: Int = 0,
     var traderSkill: Int = 0,
-    var engineerSkill: Int = 0): Parcelable {
+    var engineerSkill: Int = 0) {
 
     lateinit var system: SolarSystem
+    lateinit var solarSystems: MutableSet<SolarSystem>
```
Comment (rohanrk) [inline app/src/main/java/com/communistutopia/spacetrader/model/Player.kt]: Why do we do this? We have a solar system variable and also a set of solar systems?

**Pull request merged**